RDMA IB: Various fixes and improvements

adriansuhov opened the pull request:
This commit adds a series of improvements and fixes:
- Added dynamic RHEL/CentOS Mellanox driver installation. It will automatically detect and download the necessary drivers.
- Changed eth check from 'ifconfig' to 'ip' to avoid using net-tools package.
- Added 'memlock unlimited' config for all distros
- Fixed Intel MPI1/RMA/NBC code. It was trying to use a variable that doesn't exist ($modified_slaves instead of $slaves)
- All test VMs will have updated hostnames. Also, the main vm will be updated with all the test clients in /etc/hosts.
- Resolve-UninitializedIB function will be run on all distros except Ubuntu. It will decrease the number of random fails where port is down and a simple reboot solves it.
- Various fixes for missing packages or files.

**These changes have been tested on RHEL 7.5, RHEL 7.6, Centos 7.6, Ubuntu 18.04 and SLES 15 by running 1-2 MPI tests / distro.**

# Test results
[LISAv2 Test Results Summary]
Test Run On           : 04/19/2019 10:26:54
ARM Image Under Test  : OpenLogic : CentOS-HPC : 7.6 : 7.6.20190327
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:36

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND           INFINIBAND-INTEL-MPI-2VM                                                          PASS                27.06
        InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS
        InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS
        InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS
        InfiniBand-Verification-2-Reboot : eth0 IP : PASS
        InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS
        InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS
        InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS
        InfiniBand-Verification-2-Reboot : IMB-RMA : PASS
        InfiniBand-Verification-2-Reboot : IMB-NBC : PASS

[LISAv2 Test Results Summary]
Test Run On           : 04/19/2019 10:12:26
ARM Image Under Test  : RedHat : RHEL : 7-RAW : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:2:12

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND           INFINIBAND-IBM-MPI-2VM                                                            PASS               122.34
        InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS
        InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS
        InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS
        InfiniBand-Verification-2-Reboot : eth0 IP : PASS
        InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS
        InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS
        InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS
        InfiniBand-Verification-2-Reboot : IMB-NBC : PASS

[LISAv2 Test Results Summary]
Test Run On           : 04/19/2019 10:12:42
ARM Image Under Test  : RedHat : RHEL : 7.5 : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:2:14

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND           INFINIBAND-IBM-MPI-2VM                                                            PASS               120.51
        InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS
        InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS
        InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS
        InfiniBand-Verification-2-Reboot : eth0 IP : PASS
        InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS
        InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS
        InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS
        InfiniBand-Verification-2-Reboot : IMB-NBC : PASS

[LISAv2 Test Results Summary]
Test Run On           : 04/19/2019 11:18:16
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:23

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND           INFINIBAND-MVAPICH-MPI-2VM                                                        PASS                 15.9
        InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS
        InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS
        InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS
        InfiniBand-Verification-2-Reboot : eth0 IP : PASS
        InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS
        InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS
        InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS
        InfiniBand-Verification-2-Reboot : IMB-RMA : PASS
        InfiniBand-Verification-2-Reboot : IMB-NBC : PASS

juhlee-microsoft commented:
Can you post the test result or pipeline job number here?

iamshital commented:
Will update the results by tomorrow.

iamshital commented:
Execution is verified
----------------------
**CentOS 75**
`08/09/2018 12:58:49 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-DEPLOYMENT-PROVISION~~~~~~~~~~`
`08/09/2018 12:58:49 : [INFO ] CURRENT - PASS    - 1`
`08/09/2018 12:58:49 : [INFO ] CURRENT - FAIL    - 0`
`08/09/2018 12:58:49 : [INFO ] CURRENT - ABORTED - 0`
`08/09/2018 12:58:49 : [INFO ] Removed Completed background job ID 67.`
`08/09/2018 12:58:49 : [INFO ] Checking background cleanup jobs..`
`08/09/2018 12:58:49 : [INFO ] Removed 0 context files.`
`08/09/2018 12:58:49 : [INFO ] Cycle Finished.. TC-31740`
`08/09/2018 12:58:50 : [INFO ] `
`Test Results Summary`
`ICA test run on 08/09/2018 12:51:22`
`ARM Image under test Openlogic : CentOS : 7.5 : latest`
`Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)`
`Total Execution Time(dd:hh:mm) 0:0:7`
`XML file: TestConfiguration`

`VERIFY-DEPLOYMENT-PROVISION Execution Time: 7.4 minutes`
`	VERIFY-DEPLOYMENT-PROVISION : PASS  `
`		  FirstBoot : PASS `
`		  FirstBoot : Call Trace Verification : PASS `
`		  Reboot : PASS `
`		  Reboot : Call Trace Verification : PASS `


`Logs can be found at .\TestResults\2018-09-08-12-51-11-6532`

**CentOS 74**
`08/09/2018 12:58:47 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-DEPLOYMENT-PROVISION~~~~~~~~~~`
`08/09/2018 12:58:47 : [INFO ] CURRENT - PASS    - 1`
`08/09/2018 12:58:47 : [INFO ] CURRENT - FAIL    - 0`
`08/09/2018 12:58:47 : [INFO ] CURRENT - ABORTED - 0`
`08/09/2018 12:58:47 : [INFO ] Removed Completed background job ID 67.`
`08/09/2018 12:58:47 : [INFO ] Checking background cleanup jobs..`
`08/09/2018 12:58:48 : [INFO ] Removed 0 context files.`
`08/09/2018 12:58:48 : [INFO ] Cycle Finished.. TC-67235`
`08/09/2018 12:58:48 : [INFO ] `
`Test Results Summary`
`ICA test run on 08/09/2018 12:51:19`
`ARM Image under test Openlogic : CentOS : 7.4 : latest`
`Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)`
`Total Execution Time(dd:hh:mm) 0:0:7`
`XML file: TestConfiguration`

`VERIFY-DEPLOYMENT-PROVISION Execution Time: 7.42 minutes`
`	VERIFY-DEPLOYMENT-PROVISION : PASS  `
`		  FirstBoot : PASS `
`		  FirstBoot : Call Trace Verification : PASS `
`		  Reboot : PASS `
`		  Reboot : Call Trace Verification : PASS `


`Logs can be found at .\TestResults\2018-09-08-12-51-06-5110`

**CentOS 71**
`08/09/2018 12:58:47 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-DEPLOYMENT-PROVISION~~~~~~~~~~`
`08/09/2018 12:58:47 : [INFO ] CURRENT - PASS    - 1`
`08/09/2018 12:58:47 : [INFO ] CURRENT - FAIL    - 0`
`08/09/2018 12:58:47 : [INFO ] CURRENT - ABORTED - 0`
`08/09/2018 12:58:47 : [INFO ] Removed Completed background job ID 67.`
`08/09/2018 12:58:47 : [INFO ] Checking background cleanup jobs..`
`08/09/2018 12:58:47 : [INFO ] Removed 0 context files.`
`08/09/2018 12:58:47 : [INFO ] Cycle Finished.. TC-82860`
`08/09/2018 12:58:47 : [INFO ] `
`Test Results Summary`
`ICA test run on 08/09/2018 12:51:16`
`ARM Image under test Openlogic : CentOS : 7.1 : latest`
`Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)`
`Total Execution Time(dd:hh:mm) 0:0:7`
`XML file: TestConfiguration`

`VERIFY-DEPLOYMENT-PROVISION Execution Time: 7.46 minutes`
`	VERIFY-DEPLOYMENT-PROVISION : PASS  `
`		  FirstBoot : PASS `
`		  FirstBoot : Call Trace Verification : PASS `
`		  Reboot : PASS `
`		  Reboot : Call Trace Verification : PASS `


`Logs can be found at .\TestResults\2018-09-08-12-51-01-8868`

**CentOS 70**
`08/09/2018 12:58:15 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-DEPLOYMENT-PROVISION~~~~~~~~~~`
`08/09/2018 12:58:15 : [INFO ] CURRENT - PASS    - 1`
`08/09/2018 12:58:15 : [INFO ] CURRENT - FAIL    - 0`
`08/09/2018 12:58:15 : [INFO ] CURRENT - ABORTED - 0`
`08/09/2018 12:58:15 : [INFO ] Removed Completed background job ID 67.`
`08/09/2018 12:58:15 : [INFO ] Checking background cleanup jobs..`
`08/09/2018 12:58:15 : [INFO ] Removed 0 context files.`
`08/09/2018 12:58:15 : [INFO ] Cycle Finished.. TC-30634`
`08/09/2018 12:58:15 : [INFO ] `
`Test Results Summary`
`ICA test run on 08/09/2018 12:51:07`
`ARM Image under test Openlogic : CentOS : 7.0 : latest`
`Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)`
`Total Execution Time(dd:hh:mm) 0:0:7`
`XML file: TestConfiguration`

`VERIFY-DEPLOYMENT-PROVISION Execution Time: 7.01 minutes`
`	VERIFY-DEPLOYMENT-PROVISION : PASS  `
`		  FirstBoot : PASS `
`		  FirstBoot : Call Trace Verification : PASS `
`		  Reboot : PASS `
`		  Reboot : Call Trace Verification : PASS `


`Logs can be found at .\TestResults\2018-09-08-12-50-52-7575`

adriansuhov committed:
RDMA IB: Various fixes and improvements

This commit adds a series of improvements and fixes:
- Added dynamic RHEL/CentOS Mellanox driver installation. It will
automatically detect and download the necessary drivers.
- Changed eth check from 'ifconfig' to 'ip' to avoid using net-tools
package.
- Added 'memlock unlimited' config for all distros
- Fixed Intel MPI1/RMA/NBC code. It was trying to use a variable that
doesn't exist ($modified_slaves instead of $slaves)
- All test VMs will have updated hostnames. Also, the main vm will be
updated with all the test clients in /etc/hosts.
- Resolve-UninitializedIB function will be run on all distros except
Ubuntu. It will decrease the number of random fails where port is down
and a simple reboot solves it.
- Various fixes for missing packages or files.

juhlee-microsoft committed:
Merge branch 'master' into bugFixes
